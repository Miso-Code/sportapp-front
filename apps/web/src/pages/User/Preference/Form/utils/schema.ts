import 'config/lang/form.ts'
import { InferType, boolean, date, object, string } from 'yup'

export const virtualAppointmentSchema = object().shape({
	date: date().required(),
	training: string().required(),
	description: string().required(),
	type: boolean().required(),
	address: string().when('type', {
		is: false,
		then: () => string().required()
	})
})

export type FormVirtualAppointmentData = InferType<
	typeof virtualAppointmentSchema
>
