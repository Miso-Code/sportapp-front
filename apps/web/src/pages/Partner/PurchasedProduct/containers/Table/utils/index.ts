import { ProductPurchased } from '@sportapp/sportapp-repository/src/business-partner/interfaces/api/product-purchased'
import { Column } from '../interfaces'

export function createData(
	id: string,
	name: string,
	email: string,
	product: string,
	status: string,
	price: number
) {
	return { id, name, email, product, status, price }
}

export function generateRows(data: ProductPurchased[]) {
	return data.map((item) =>
		createData(
			item.product_data.product_id,
			item.user_name,
			item.user_email,
			item.product_data.name,
			item.transaction_status,
			item.product_data.price
		)
	)
}

export const columns: readonly Column[] = [
	{ id: 'id', label: 'paymentPurchased.table.header.0', minWidth: 170 },
	{ id: 'name', label: 'paymentPurchased.table.header.1', minWidth: 100 },
	{
		id: 'email',
		label: 'paymentPurchased.table.header.2',
		minWidth: 170
	},
	{
		id: 'product',
		label: 'paymentPurchased.table.header.3',
		minWidth: 100
	},
	{
		id: 'status',
		label: 'paymentPurchased.table.header.4',
		align: 'left',
		minWidth: 100
	},
	{
		id: 'price',
		label: 'paymentPurchased.table.header.5',
		minWidth: 170,
		format: (value: number) => value.toFixed(2)
	}
]

export function formatCurrency(value: number) {
	return value.toFixed(2).replace(/\d(?=(\d{3})+\.)/g, '$&,')
}
