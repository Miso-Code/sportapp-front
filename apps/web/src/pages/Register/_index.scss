.register {
	@apply flex;

	min-height: 100vh;

	.title {
		@apply font-bold mb-7;
	}

	.navigation {
		@apply mt-4;
	}

	.section {
		&-main {
			@apply flex-1 p-2 flex flex-col items-center justify-center;
		}

		&-secondary {
			@apply w-1/2 lg:block hidden;
		}
	}

	.container-image {
		@apply w-full h-full;
	}

	.image-register {
		@apply object-cover object-center w-full h-full;
	}

	.card {
		&-register {
			@apply w-full p-4 pb-4;

			min-width: 300px;
			max-width: 466px;
		}

		&-title {
			@apply w-full text-center mb-6;
		}
	}
}
