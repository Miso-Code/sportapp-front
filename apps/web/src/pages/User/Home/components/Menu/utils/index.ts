import { SetStateAction } from 'react'

export const handleEdit = (
	index: number,
	isEditing: boolean[],
	setIsEditing: (value: SetStateAction<boolean[]>) => void
) => {
	const newIsEditing = isEditing.map((value, i) => {
		if (i === index) return !value
		return value
	})
	setIsEditing(newIsEditing)
}
