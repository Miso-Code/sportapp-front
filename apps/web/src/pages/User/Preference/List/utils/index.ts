export function createData(
	id: string,
	dateAndTime: string,
	type: string,
	trainingName: string,
	location?: string
) {
	return { id, dateAndTime, type, location, trainingName }
}
