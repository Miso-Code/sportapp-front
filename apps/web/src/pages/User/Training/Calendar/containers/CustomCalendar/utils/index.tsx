import { EventWrapperCustomProps } from '../interfaces'

export const components = {
	eventWrapper: (props: EventWrapperCustomProps) => {
		const { event } = props
		return (
			<div
				className={`event-wrapper event-custom-type-${event.type} ${event.custom_class ?? ''}`}>
				{props.children}
			</div>
		)
	}
}
