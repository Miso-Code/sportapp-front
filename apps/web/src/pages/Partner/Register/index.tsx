import SecondarySection from '@/components/SecondarySection'
import TransitionAlert from '@/components/TransitionAlert'
import RegisterPartnerContainer from '@/containers/Partner/Register'
import { FormData } from '@/containers/Partner/Register/utils/schema'
import { Button } from '@mui/material'
import Paper from '@mui/material/Paper'
import Typography from '@mui/material/Typography'
import { RegisterBusinessPartnerRequest } from '@sportapp/sportapp-repository/src/business-partner/interfaces/api/register'
import { usePartnerAuthStore } from '@sportapp/stores/src/partner/auth'
import registerImage from 'assets/images/login-wallpaper.jpg'
import { useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { useNavigate } from 'react-router-dom'
import './_index.scss'

export default function Register() {
	const [alertState, setAlert] = useState(false)
	const navigate = useNavigate()

	const { t } = useTranslation()
	const { register, logout } = usePartnerAuthStore()
	const { loading, error } = usePartnerAuthStore()

	const handleFirstSubmit = async (data: FormData) => {
		const payload: RegisterBusinessPartnerRequest = {
			email: data.email.toLowerCase(),
			password: data.password,
			business_partner_name: data.companyName
		}

		const result = await register(payload)

		if (result) {
			navigate('/partner/home')
		} else {
			setAlert(true)
		}
	}

	const handleGoToLogin = () => {
		console.log('navigate', navigate)
		navigate('/partner/login')
	}

	useEffect(() => {
		logout()
	}, [logout])

	return (
		<>
			<div className='register'>
				<main className='section-main'>
					<Typography className='title' variant='h1'>
						{t('app.name')}
					</Typography>
					<Paper variant='outlined' className={`card-register`}>
						<Typography className='card-title' variant='h6'>
							{t('register.partner.title')}
						</Typography>
						<RegisterPartnerContainer
							onHandleSubmit={handleFirstSubmit}
						/>

						<div className='flex items-center justify-center navigation'>
							<Typography
								className='card-subtitle'
								variant='subtitle1'>
								{t('register.login.question')}
							</Typography>
							<Button
								type='button'
								disabled={loading}
								onClick={handleGoToLogin}
								variant='text'
								title={t('register.button')}>
								{t('register.login.default')}
							</Button>
						</div>
					</Paper>
				</main>

				<SecondarySection
					image={registerImage}
					altImage={t('register.image')}
				/>
			</div>
			<TransitionAlert
				containerClassName='alert-register-container'
				isOpen={alertState}
				handleClose={setAlert}
				message={t(error ?? 'errors.register.base')}
				severity='error'
			/>
		</>
	)
}
