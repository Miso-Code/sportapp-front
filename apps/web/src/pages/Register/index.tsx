import RegisterFullContainer from '@/containers/Register/Full'
import { Button } from '@mui/material'
import Paper from '@mui/material/Paper'
import Typography from '@mui/material/Typography'
import registerImage from 'assets/images/login-wallpaper.jpg'
import 'pages/Register/_index.scss'
import { useTranslation } from 'react-i18next'

export default function Register() {
	const { t } = useTranslation()
	return (
		<div className='register'>
			<main className='section-main'>
				<Typography className='title' variant='h1'>
					{t('app.name')}
				</Typography>
				<Paper className='card-register'>
					<Typography className='card-title' variant='h6'>
						{t('register.default')}
					</Typography>
					{/* <RegisterContainer onHandleSubmit={console.log} /> */}
					<RegisterFullContainer onHandleSubmit={console.log} />
					<Button
						fullWidth
						type='button'
						variant='text'
						title={t('register.button')}
						className='navigation'>
						{t('register.login')}
					</Button>
				</Paper>
			</main>
			<section className='section-secondary'>
				<figure className='container-image'>
					<img
						className='image-register'
						src={registerImage}
						alt={t('register.image')}
					/>
				</figure>
			</section>
		</div>
	)
}
