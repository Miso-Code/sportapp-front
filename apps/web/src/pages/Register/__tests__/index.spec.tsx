import { FormData } from '@/containers/Register/Default/utils/schema'
import { FormData as FormDataFull } from '@/containers/Register/Full/utils/schema'
import { useAuthStore } from '@sportapp/stores/src/auth'
import { render, RenderResult } from '@testing-library/react'
import Register from 'pages/Register'
import { useState } from 'react'

jest.mock(
	'@/containers/Register',
	() =>
		({
			step,
			onHandleFirstSubmit,
			onHandleSecondSubmit
		}: {
			step: number
			onHandleFirstSubmit: (data: FormData) => void
			onHandleSecondSubmit: (data: FormDataFull) => void
		}) =>
			(
				<div>
					RegisterContainer-{step}
					<button
						data-testid='onHandleFirstSubmit'
						onClick={() => {
							const data: FormData = {
								email: 'email',
								password: 'password',
								name: 'name',
								lastName: 'lastName'
							}
							onHandleFirstSubmit(data)
						}}></button>
					<button
						data-testid='onHandleSecondSubmit'
						onClick={() => {
							const data: FormDataFull = {
								birthday: new Date().toISOString(),
								documentNumber: 'documentNumber',
								documentType: 'documentType',
								email: 'email',
								gender: 'M',
								lastName: 'lastName',
								name: 'name',
								password: 'password',
								nationality: {
									city: 'city',
									country: 'country'
								},
								residence: {
									country: 'country',
									city: 'city',
									lengthOfStay: 'lengthOfStay'
								}
							}
							onHandleSecondSubmit(data)
						}}></button>
				</div>
			)
)

jest.mock('react', () => {
	const ActualReact = jest.requireActual('react')
	return {
		...ActualReact,
		useState: jest.fn(() => [0, jest.fn()])
	}
})

jest.mock('@sportapp/stores/src/auth', () => ({
	useAuthStore: () => ({
		login: jest.fn().mockReturnValueOnce(true).mockReturnValueOnce(false),
		logout: jest.fn(),
		register: jest.fn().mockResolvedValue(true),
		registerFull: jest.fn().mockResolvedValue(true),
		loading: false,
		error: null
	})
}))

jest.mock('@/components/TransitionAlert', () => () => (
	<div>TransitionAlert</div>
))

describe('Register', () => {
	let wrapper: RenderResult

	beforeEach(() => {
		wrapper = render(<Register />)
	})

	afterEach(() => {
		wrapper.unmount()
	})

	it('should render correctly', () => {
		expect(wrapper).toMatchSnapshot()
	})

	it('should call onHandleFirstSubmit', () => {
		;(useState as jest.Mock).mockImplementationOnce(() => [0, jest.fn()])
		wrapper.rerender(<Register />)
		wrapper.getByTestId('onHandleFirstSubmit').click()
		expect(useState).toHaveBeenCalledWith(0)
	})

	it('should call onHandleSecondSubmit', () => {
		;(useState as jest.Mock).mockImplementationOnce(() => [1, jest.fn()])
		wrapper.rerender(<Register />)
		wrapper.getByTestId('onHandleSecondSubmit').click()
		expect(useState).toHaveBeenCalledWith(0)
	})

	it('should not call onHandleSubmit', () => {
		;(useState as jest.Mock).mockImplementationOnce(() => [2, jest.fn()])
		wrapper.rerender(<Register />)
		expect(useState).toHaveBeenCalledWith(0)
	})

	it.skip('should call handleSubmit', () => {
		;(useState as jest.Mock).mockImplementationOnce(() => [1, jest.fn()])
		;(useAuthStore as unknown as jest.Mock).mockReturnValue({
			login: jest.fn().mockResolvedValue(true)
		})
		wrapper.rerender(<Register />)
		wrapper.getByTestId('onHandleSecondSubmit').click()
		expect(useState).toHaveBeenCalledWith(0)
	})
})
