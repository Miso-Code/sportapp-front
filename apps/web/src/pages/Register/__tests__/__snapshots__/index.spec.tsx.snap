// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="register"
      >
        <main
          class="section-main"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 title css-o2w69a-MuiTypography-root"
          >
            app.name
          </h1>
          <div
            class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded card-register false css-172kgz0-MuiPaper-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 card-title css-2ulfj5-MuiTypography-root"
            >
              register.default
            </h6>
            <div>
              RegisterContainer-
              0
              <button
                data-testid="onHandleFirstSubmit"
              />
              <button
                data-testid="onHandleSecondSubmit"
              />
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-fullWidth navigation css-10vjzfb-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              title="register.button"
              type="button"
            >
              register.go
            </button>
          </div>
        </main>
        <section
          class="section-secondary"
        >
          <figure
            class="container-image"
          >
            <img
              alt="register.image"
              class="image-register"
              src="mocked-file-content"
            />
          </figure>
        </section>
      </div>
      <div>
        TransitionAlert
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="register"
    >
      <main
        class="section-main"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 title css-o2w69a-MuiTypography-root"
        >
          app.name
        </h1>
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded card-register false css-172kgz0-MuiPaper-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 card-title css-2ulfj5-MuiTypography-root"
          >
            register.default
          </h6>
          <div>
            RegisterContainer-
            0
            <button
              data-testid="onHandleFirstSubmit"
            />
            <button
              data-testid="onHandleSecondSubmit"
            />
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-fullWidth navigation css-10vjzfb-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            title="register.button"
            type="button"
          >
            register.go
          </button>
        </div>
      </main>
      <section
        class="section-secondary"
      >
        <figure
          class="container-image"
        >
          <img
            alt="register.image"
            class="image-register"
            src="mocked-file-content"
          />
        </figure>
      </section>
    </div>
    <div>
      TransitionAlert
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
